djangocms_content_expiry/monkeypatch/models.py

```diff
@@ -0,0 +1,68 @@
+from dateutil.relativedelta import relativedelta
+
+from djangocms_versioning import models
+
+from djangocms_content_expiry.models import ContentExpiry
+
+
+def new_save(old_save):
+    """
+    Override the Versioning save method to add the expiration date
+    """
+    def inner(version, **kwargs):
+        old_save(version, **kwargs)
+
+        # Check if version has content expiry added
+        new_version = _version_has_expiry(version)
+        if new_version is None:
```

Comment:
I would propose to first start simple, we don't yet have a record so let's create one.

```suggestion
        content_expiry = getattr(version, "contentexpiry", None)
        # Content expiry object exists so copying the original expiry object
        if not content_expiry:
            ContentExpiry.objects.create(
                version=version,
                created=version.created,
                created_by=version.created_by,
                expires=_get_future_expire_date(datetime.now())
            )
            return version 
```